In [16]:
import pandas as pd
import numpy as np
import os
import functools as ft
import matplotlib.pyplot as plt
import sklearn

In [17]:
charlie = pd.read_csv('employment_data(2).csv')
yash = pd.read_csv('prag_merged_data.csv')
kyle = pd.read_csv('readyForGroup.csv')
kyle2 = pd.read_csv('censusHousing.csv')
charlie = charlie.rename(columns={'Year': 'year'})

In [18]:
charlie

year   FIPS  Labor Force Participation Rate(ages 16 and over)  \
0     2022   1001                                              59.6   
1     2022   1003                                              59.0   
2     2022   1005                                              44.0   
3     2022   1007                                              50.0   
4     2022   1009                                              56.0   
...    ...    ...                                               ...   
3217  2022  72145                                              44.1   
3218  2022  72147                                              34.6   
3219  2022  72149                                              42.8   
3220  2022  72151                                              37.8   
3221  2022  72153                                              39.3   

      Unemployment Rate(ages 16 and over) Below poverty line in last 12 months  
0                                     2.8                                 25.9  
1                                     3.4                                 40.1  
2                                     5.8                                   25  
3                                     8.7                                 24.8  
4                                     6.0                                   24  
...                                   ...                                  ...  
3217                                 16.3                                 32.5  
3218                                 13.1                                 27.5  
3219                                 13.9                                 28.3  
3220                                 11.1                                 30.8  
3221                                 13.5                                 23.1  

[3222 rows x 5 columns]

In [19]:
kyle.head(1000)

year   FIPS popDensityType  analysisValue  % Uninsured  \
0    2022   1001          Urban            588     9.351239   
1    2022   1003          Rural            608    10.906774   
2    2022   1005          Rural            747    12.991473   
3    2022   1007          Urban            593    10.993250   
4    2022   1009          Urban            601    14.271027   
..    ...    ...            ...            ...          ...   
995  2022  21019          Urban            690     7.080815   
996  2022  21021          Rural            602     6.563048   
997  2022  21023          Urban            559     7.351431   
998  2022  21025          Rural            880     7.777445   
999  2022  21027          Rural            534     8.876034   

     Primary Care Physicians Rate  Dentist Rate  Mental Health Provider Rate  \
0                        44.74753      33.84095                     37.40315   
1                        68.98591      48.84708                     99.43869   
2                        36.45791      36.60173                     24.40115   
3                        58.05126      27.10517                     27.10517   
4                        22.48124      19.00517                     17.27742   
..                            ...           ...                          ...   
995                     109.16563      70.94333                    778.22685   
996                      66.53360      79.03316                    250.27168   
997                      24.08768      24.13710                     24.13710   
998                      55.42359      15.93625                    247.01195   
999                      24.41764      14.60778                     97.38521   

     percentVacantHousing  percentOwnerOccupied  percentRenterOccupied  
0                     NaN                   NaN                    NaN  
1                     NaN                   NaN                    NaN  
2                     NaN                   NaN                    NaN  
3                     NaN                   NaN                    NaN  
4                     NaN                   NaN                    NaN  
..                    ...                   ...                    ...  
995                   NaN                   NaN                    NaN  
996                   NaN                   NaN                    NaN  
997                   NaN                   NaN                    NaN  
998                   NaN                   NaN                    NaN  
999                   NaN                   NaN                    NaN  

[1000 rows x 11 columns]

In [20]:
yash

FIPS  year  Less than 50k  50k to 99,999  100k to 149,999  Over 150k  \
0      1003  2019          0.451          0.274            0.160      0.115   
1      1015  2019          0.512          0.309            0.129      0.050   
2      1043  2019          0.489          0.373            0.086      0.053   
3      1049  2019          0.557          0.296            0.110      0.037   
4      1051  2019          0.391          0.324            0.163      0.122   
...     ...   ...            ...            ...              ...        ...   
2524  72113  2022          0.853          0.113            0.026      0.006   
2525  72127  2022          0.715          0.167            0.060      0.059   
2526  72135  2022          0.540          0.343            0.078      0.037   
2527  72137  2022          0.746          0.184            0.045      0.025   
2528  72139  2022          0.688          0.200            0.043      0.070   

      Total not grad HS  Total grad HS  Total Associates  Total Bachelors  \
0                   NaN            NaN               NaN              NaN   
1                   NaN            NaN               NaN              NaN   
2                   NaN            NaN               NaN              NaN   
3                   NaN            NaN               NaN              NaN   
4                   NaN            NaN               NaN              NaN   
...                 ...            ...               ...              ...   
2524           0.103954       0.449606          0.069200         0.366229   
2525           0.070624       0.252405          0.080180         0.548250   
2526           0.052730       0.660033          0.113069         0.421429   
2527           0.063966       0.625110          0.109130         0.385442   
2528           0.037868       0.569296          0.059897         0.415716   

      ...  Female 55 yrs and over                   county        state  \
0     ...                     NaN           Baldwin County      ALABAMA   
1     ...                     NaN           Calhoun County      ALABAMA   
2     ...                     NaN           Cullman County      ALABAMA   
3     ...                     NaN            DeKalb County      ALABAMA   
4     ...                     NaN            Elmore County      ALABAMA   
...   ...                     ...                      ...          ...   
2524  ...                0.227962          Ponce Municipio  PUERTO RICO   
2525  ...                0.255485       San Juan Municipio  PUERTO RICO   
2526  ...                0.184512       Toa Alta Municipio  PUERTO RICO   
2527  ...                0.301076       Toa Baja Municipio  PUERTO RICO   
2528  ...                0.169159  Trujillo Alto Municipio  PUERTO RICO   

      urban  analysis_value    State   County  PCP per 1000  \
0     Rural           602.0  Alabama  Baldwin           NaN   
1     Urban           658.0  Alabama  Calhoun           NaN   
2     Rural           629.0  Alabama  Cullman           NaN   
3     Rural           669.0  Alabama   DeKalb           NaN   
4     Urban           645.0  Alabama   Elmore           NaN   
...     ...             ...      ...      ...           ...   
2524  Urban           406.0      NaN      NaN           NaN   
2525  Urban           373.0      NaN      NaN           NaN   
2526  Urban           347.0      NaN      NaN           NaN   
2527  Urban           349.0      NaN      NaN           NaN   
2528  Urban           385.0      NaN      NaN           NaN   

      Dentists per 1000  MHP per 1000  
0              49.85232     118.21522  
1              64.50038      37.40315  
2              48.33545      99.43869  
3              25.13370      24.40115  
4              14.69202      27.10517  
...                 ...           ...  
2524                NaN      75.63459  
2525                NaN     132.31631  
2526                NaN       8.05575  
2527                NaN     285.68524  
2528                NaN      27.658

In [21]:
kyle2

FIPS  year  percentVacantHousing  percentOwnerOccupied  \
0      1015  2022             14.430422             61.537597   
1      1043  2022              9.843832             70.724187   
2      1049  2022             14.708916             62.033396   
3      1051  2022              8.876642             69.897536   
4      1055  2022             16.132824             61.441585   
...     ...   ...                   ...                   ...   
3361  72113  2018             18.145760             55.642506   
3362  72127  2018             22.726533             41.961077   
3363  72135  2018             18.583702             65.758298   
3364  72137  2018             25.034487             57.046787   
3365  72139  2018             20.819963             53.403427   

      percentRenterOccupied  
0                 24.031980  
1                 19.431981  
2                 23.257688  
3                 21.225822  
4                 22.425591  
...                     ...  
3361              26.211734  
3362              35.312390  
3363              15.657999  
3364              17.918726  
3365              25.776610  

[3366 rows x 5 columns]

In [25]:
groupDataSet = pd.merge(yash, charlie, on=['year', 'FIPS'], how='left')

groupDataSet = pd.merge(groupDataSet, kyle2, on=['year', 'FIPS'], how='left')

In [26]:
groupDataSet

FIPS  year  Less than 50k  50k to 99,999  100k to 149,999  Over 150k  \
0      1003  2019          0.451          0.274            0.160      0.115   
1      1015  2019          0.512          0.309            0.129      0.050   
2      1043  2019          0.489          0.373            0.086      0.053   
3      1049  2019          0.557          0.296            0.110      0.037   
4      1051  2019          0.391          0.324            0.163      0.122   
...     ...   ...            ...            ...              ...        ...   
2524  72113  2022          0.853          0.113            0.026      0.006   
2525  72127  2022          0.715          0.167            0.060      0.059   
2526  72135  2022          0.540          0.343            0.078      0.037   
2527  72137  2022          0.746          0.184            0.045      0.025   
2528  72139  2022          0.688          0.200            0.043      0.070   

      Total not grad HS  Total grad HS  Total Associates  Total Bachelors  \
0                   NaN            NaN               NaN              NaN   
1                   NaN            NaN               NaN              NaN   
2                   NaN            NaN               NaN              NaN   
3                   NaN            NaN               NaN              NaN   
4                   NaN            NaN               NaN              NaN   
...                 ...            ...               ...              ...   
2524           0.103954       0.449606          0.069200         0.366229   
2525           0.070624       0.252405          0.080180         0.548250   
2526           0.052730       0.660033          0.113069         0.421429   
2527           0.063966       0.625110          0.109130         0.385442   
2528           0.037868       0.569296          0.059897         0.415716   

      ...   County  PCP per 1000  Dentists per 1000  MHP per 1000  \
0     ...  Baldwin           NaN           49.85232     118.21522   
1     ...  Calhoun           NaN           64.50038      37.40315   
2     ...  Cullman           NaN           48.33545      99.43869   
3     ...   DeKalb           NaN           25.13370      24.40115   
4     ...   Elmore           NaN           14.69202      27.10517   
...   ...      ...           ...                ...           ...   
2524  ...      NaN           NaN                NaN      75.63459   
2525  ...      NaN           NaN                NaN     132.31631   
2526  ...      NaN           NaN                NaN       8.05575   
2527  ...      NaN           NaN                NaN     285.68524   
2528  ...      NaN           NaN                NaN      27.65869   

      Labor Force Participation Rate(ages 16 and over)  \
0                                                  NaN   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4                                                  NaN   
...                                                ...   
2524                                              37.7   
2525                                              52.3   
2526                                              51.3   
2527                                              52.3   
2528                                              53.0   

      Unemployment Rate(ages 16 and over)  \
0                                     NaN   
1                                     NaN   
2                                     NaN   
3                                     NaN   
4                                     NaN   
...                                   ...   
2524                                 12.7   
2525                                 15.7   
2526                                  8.6   
2527                                 11.3   
2528                                  6.5   

      Below poverty line in last 12 months  percentVacantHousing  

In [27]:
groupDataSet.to_csv('groupDataSet.csv', index=False)

FIPS- A code used to identify counties
Year-A way to identify which year data came from
Less than 50k-a percentage of people that have an income less than
50k 
50k to 99,999- a percentage of people that have an income between
50k and $99,999
100k to 149,999- a percentage of people that have an income between
100k and $149,999
Over 150k- a percentage of people that have an income over 150k

The above income statistics could prove useful as our hypothesis is that higher income individuals will
have better access to care.

Total not grad HS- a percentage of people that haven’t graduated high
school
Total grad HS- A percentage of people that have graduated high school,
but have not received degrees
Total Associates- percentage of people that have received associate
degrees
Total Bachelors- percentage of people that have received bachelor
degrees
Total Professional or Graduate Degrees- percentage of people that
have received professional or graduate degrees

The above education statistics could prove useful as we hypothesize that higher educated areas are more likely
to have high access to care

Total White alone- a percentage of people that are mainly/ only white
Total Black or African American alone- a percentage of people that are
mainly/ only black
Total American Indian and Alaska Native alone- a percentage of
people that are mainly/ only American Indian or an Alaska native
Total Asian alone- a percentage of people that are mainly/ only asian
Total Native Hawaiian and Other Pacific Islander alone- a percentage
of people that are mainly/ only a native Hawaiian or other pacific
islander
Total Some other race alone- a percentage of people that are mainly/
only a different race not mentioned above
Total Two or more races- a percentage of people that are at least 2 of
the mentioned races

The above race statistics could prove useful in understanding if access to care is a racially biased issue.

Under 25 years- The percent of people under 25 years old that have
insurance
26 to 54 years- The percent of people between 26 and 54 years old that
have insurance
Over 55 years- The percent of people over 55 years old that have
insurance
Male under 20 yrs- the percent of people under 20 years that are also
male
Male 20 to 54 yrs- the percent of people 20 to 54 years that are also
male
Male 55 yrs and over- the percent of people over 55 years that are also
male
Female under 20 yrs- the percent of people under 20 years that are
also female
Female 20 to 54 yrs- the percent of people under people 20 to 54 years
old that are also female.
Female 55 yrs and over- the percent of people over 55 years that are
also female.

The above age statsitcs could help us determine the age and gender range most likely to have poor/good access to care.

County- The county corresponding to the FIPS code
State-The state that contains the counties
Urban- signifies if the county is mainly rural, suburban, or urban
analysis_value- a number that quantifies the quality of a emergency
department
PCP per 1000- the primary care provider rate per 1000
Dentists per 1000- the dentist rate per 1000
MHP per 1000- the mental health provider rate per 1000
percentVacantHousing - This could prove useful as vacant housing could indicate a unique socioeconomic situation in that county which could have an impact on access to care
percentOwnerOccupied - More owners occuping homes is indicative of wealth which could have an impact. Also more stable housing means more likely to be able to schedule proper appointments
percentRenterOccupied - pretty much the same as above but reversed.